sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs

```diff
@@ -45,99 +37,41 @@ public async Task VerifyImdsRequestMockAsync()
 
                 MockRequest request = mockTransport.Requests[0];
 
                 string query = request.Uri.Query;
 
                 Assert.IsTrue(query.Contains("api-version=2018-02-01"));
-
                 Assert.IsTrue(query.Contains($"resource={Uri.EscapeDataString(ScopeUtilities.ScopesToResource(MockScopes.Default))}"));
-
                 Assert.IsTrue(request.Headers.TryGetValue("Metadata", out string metadataValue));
-
+                Assert.IsTrue(query.Contains($"client_id=mock-client-id"));
                 Assert.AreEqual("true", metadataValue);
             }
         }
 
         [NonParallelizable]
         [Test]
-        public async Task VerifyImdsRequestWithClientIdMockAsync()
+        [TestCase(400)]
+        [TestCase(502)]
+        public void VerifyImdsRequestHandlesFailedRequestWithCredentialUnavailableExceptionMockAsync(int responseCode)
         {
             using (new TestEnvVar(new () { { "MSI_ENDPOINT", null }, { "MSI_SECRET", null } }))
             {
-                var response = new MockResponse(200);
-
-                var expectedToken = "mock-msi-access-token";
-
-                response.SetContent($"{{ \"access_token\": \"{expectedToken}\", \"expires_on\": \"3600\" }}");
-
+                var response = CreateMockResponse(responseCode, ExpectedToken);
                 var mockTransport = new MockTransport(response);
-
                 var options = new TokenCredentialOptions() { Transport = mockTransport };
-
+                options.Retry.MaxRetries = 0;
                 var pipeline = CredentialPipeline.GetInstance(options);
 
                 var client = new MockManagedIdentityClient(pipeline, "mock-client-id") { ManagedIdentitySourceFactory = () => new ImdsManagedIdentitySource(pipeline, "mock-client-id") };
 
                 ManagedIdentityCredential credential = InstrumentClient(new ManagedIdentityCredential(client));
 
-                AccessToken actualToken = await credential.GetTokenAsync(new TokenRequestContext(MockScopes.Default));
-
-                Assert.AreEqual(expectedToken, actualToken.Token);
-
-                MockRequest request = mockTransport.Requests[0];
-
-                string query = request.Uri.Query;
-
-                Assert.IsTrue(query.Contains("api-version=2018-02-01"));
+                var ex = Assert.ThrowsAsync<CredentialUnavailableException>(async () => await credential.GetTokenAsync(new TokenRequestContext(MockScopes.Default)));
 
-                Assert.IsTrue(query.Contains($"resource={Uri.EscapeDataString(ScopeUtilities.ScopesToResource(MockScopes.Default))}"));
-
-                Assert.IsTrue(query.Contains($"client_id=mock-client-id"));
-
-                Assert.IsTrue(request.Headers.TryGetValue("Metadata", out string metadataValue));
-
-                Assert.AreEqual("true", metadataValue);
-            }
-        }
-
-        [NonParallelizable]
-        [Test]
-        public async Task VerifyAppService2017RequestMockAsync()
```

Comment:
Was this test a duplicate?

Reply:
Yes, dupe of `VerifyAppService2017RequestWithClientIdMockAsync`

Reply:
It looks like `VerifyAppService2017RequestWithClientIdMockAsync` always requests a user assigned id. I think we should test with this parameter an without. Perhaps we can add a parameter to `VerifyAppService2017RequestWithClientIdMockAsync` so we can test with a clientId and with specifying null for the clientId?